src/Domain/DonationService.php

```diff
@@ -710,14 +719,21 @@ public function findAllCompleteForCustomerAsAPIModels(StripeCustomerId $stripeCu
     /**
      * @throws PaymentIntentNotSucceeded
      * */
-    public function confirmDonationWithSavedPaymentMethod(Donation $donation, StripePaymentMethodId $paymentMethod): void
+    public function confirmDonationWithSavedPaymentMethod(Donation $donation, StripePaymentMethodId $paymentMethodId): void
     {
         $paymentIntentId = $donation->getTransactionId();
+
+        $this->updateDonationFeesFromPaymentMethodId($donation, $paymentMethodId);
```

Comment:
maybe better to make this line something like `$donation->setPaymentMethod($paymentMethod)`, and have the implementation of that set the fee as a side effect? Maybe also move that into the two callers of this method (`\MatchBot\Domain\RegularGivingService::setupNewMandate` and `\MatchBot\Domain\DonationService::confirmPreAuthorized`)

And then we'd remove the $paymentMethodId param from this method, and either call a new function `$donation->getPaymentMethod()` (or `$donation->getPaymentMethodId()` ) or possibly `$donation->getStripeConfirmationPayload()`. The latter of course would need to account for the more common case of having a confirmation token instead of a payment method.